qemu/tests/qemu_guest_agent_snapshot.py

```diff
@@ -48,8 +49,11 @@ def _action_after_fsfreeze(self, *args):
             image_params = self.params.object_params(image_tag)
             snapshot_test = LiveSnapshot(self.test, self.params,
                                          self.env, image_tag)
+            snapshot_create = BlockDevSnapshotTest(self.test, self.params,
+                                         self.env)
             error_context.context("Creating snapshot", logging.info)
-            snapshot_test.create_snapshot()
+            snapshot_create.prepare_snapshot_file()
```

Comment:
We add a snapshot file, we have to remove it after test, so please override the cleanup function derived from parent, and remove the snapshot file by:

snapshot_test.snapshot_image.remove()

Reply:
snapshot_test.snapshot_image.remove() -> snapshot_create.snapshot_image.remove()

Reply:
finish editing

Reply:
I mean override cleanup function derived from QemuGuestAgentBasicCheckWin, add function snapshot_create.snapshot_image.remove(), like
def cleanup(self, ....):
super(...).cleanup()
snapshot_create.snapshot_image.remove()

Given that we hit a failure in check_snapshot, then the remove function in line73 never called.
btw: It's OK to call snapshot_create.snapshot_image.remove() when the image doesn't exist, for vt will make a check before file remove

Reply:
any update for this? any failure in create_snapshot or check_snapshot occurred, snapshot_image cannot be removed

Reply:
updated to use cleanup from father class